src/model.rs

```diff
@@ -146,7 +146,7 @@ pub struct JsonOutput {
 
 #[derive(Clone, Serialize, Deserialize, Debug, Default)]
 pub struct Playlist {
-    pub entries: Option<Vec<SingleVideo>>,
+    pub entries: Option<Vec<Option<SingleVideo>>>,
```

Comment:
Is it necessary to have nested options here?

Reply:
Yes, if a video fails such as a "Premium" video, it puts "null" in the list of entries, which parses to `Option<SingleVideo>`

Reply:
Hmm, I see. That's pretty awkward from an API perspective, unfortunately. I'll merge it for now and look into improvements before a new release. (also, sorry for taking so long)

Reply:
A custom serializer like the one in the file already might be able to filter out none from the entries, I'm not familiar with serde custom serializers